src/main/java/it/pagopa/transactions/projections/handlers/TransactionsActivationProjectionHandler.java

```diff
@@ -24,23 +26,26 @@
 	public Mono<TransactionActivated> handle(TransactionActivatedEvent event) {
 		TransactionActivatedData data = event.getData();
 		TransactionId transactionId = new TransactionId(UUID.fromString(event.getTransactionId()));
-		PaymentToken paymentToken = new PaymentToken(event.getPaymentToken());
-		RptId rptId = new RptId(event.getRptId());
-		TransactionDescription description = new TransactionDescription(data.getDescription());
-		TransactionAmount amount = new TransactionAmount(data.getAmount());
+		List<NoticeCode> noticeCodeList = event.getNoticeCodes().stream().map(noticeCode ->
+			new NoticeCode(
+					new PaymentToken(noticeCode.getPaymentToken()),
+					new RptId(noticeCode.getRptId()),
+					new TransactionAmount(data.getNoticeCodes().stream().filter(noticeCode1 -> noticeCode1.getRptId().equals(noticeCode.getRptId())).findFirst().get().getAmount()),
+					new TransactionDescription(data.getNoticeCodes().stream().filter(noticeCode1 -> noticeCode1.getRptId().equals(noticeCode.getRptId())).findFirst().get().getDescription()))
```

Comment:
why not just noticeCode.getAmount() and noticeCode.getDescription()?

Reply:
They comes from main branch. This datas come from TransactionEventActivatedData so they come from the same origin.